pkg/scm/gitlab/context.go

```diff
@@ -158,3 +164,47 @@ func (c *Context) HasExecutedActionGroup(group string) bool {
 
 	return ok
 }
+
+// GetCodeOwners returns the eligible code owners for the merge request
+//
+// This is based on the elibible approvers in the rules of the merge request approval
+// state api.
+func (c *Context) GetCodeOwners() []scm.Actor {
+	owners := []scm.Actor{}
+
+	for _, rule := range c.MergeRequest.ApprovalState.Rules {
+		// Multiple code owner paths could be matched when sections are used, so
+		// flatten the list of eligible approvers
+		if rule.Type.String() != "CODE_OWNER" {
+			continue
+		}
+
+		// Note that the anyone who has authored a commit in the MR won't be considered
+		// an eligible approver as part of the GitLab API response.
+		if rule.EligibleApprovers != nil {
+			for _, user := range rule.EligibleApprovers {
+				if user.Bot == true {
+					continue
+				}
+
+				if containsActorByID(owners, user.ID) {
+					continue
+				}
+
+				owners = append(owners, user.ToActor())
+			}
+		}
+	}
+
+	return owners
+}
+
+func containsActorByID(actors []scm.Actor, id string) bool {
```

Comment:
Thought: should we move `[]scm.Actor` to a custom type `scm.Actors = []scm.Actor`  and add `Add()`/ `Has()` methods to that, so the API is cleaner?

Reply:
I like that, will update